gzap.go

```diff
@@ -24,35 +22,23 @@ var lowPriority = zap.LevelEnablerFunc(func(lvl zapcore.Level) bool {
 	return lvl < zapcore.ErrorLevel
 })
 
-// logInitializer represents a function that initializes
-type logInitializer func(cfg Config) error
-
-// envToLogInitializerMapping represents the different type of log initializers
-// to their correlating env level.
-var envToLogInitializerMapping = map[int]logInitializer{
-	testEnv:    setTestLogger,
-	devEnv:     setDevelopmentLogger,
-	stagingEnv: setStagingLogger,
-	prodEnv:    setProductionLogger,
-}
-
 // InitLogger initializes a global Logger based upon your env configurations.
 func InitLogger() error {
-	return initLogger(Config{})
+	return initLogger(&EnvConfig{})
 }
 
 func initLogger(cfg Config) error {
-	env, err := cfg.getGraylogEnv()
-	if err != nil {
-		return err
+	if cfg.getIsTestEnv() {
+		return setTestLogger(cfg)
 	}
 
-	logInitalizer, ok := envToLogInitializerMapping[env]
-	if !ok {
-		return errors.New(envNotSetErrorString)
+	// Check if graylog host is defined.
+	graylogHost := cfg.getGraylogHost()
+	if graylogHost != "" {
+		return setGELFAndConsoleLogger(cfg)
```

Comment:
Why not have a `setGelfLogger` and reuse the logic for `setConsoleLogger`, then compose calls to the two to achieve logging of both kinds?

Reply:
I personally think it's better to separate the two for a few reasons. 

For one it's kind of hard to compose the logger dynamically. Second, I think since we use either a console logger, or both a console logger and Graylog logger, so it helps illustrate the point a bit easier as to when one is used and one isn't.

Reply:
How would one 'unit test' something that does not do a unit of work? Anytime a function has a conjunction in its name, that suggests to me a design issue. Functions should have one thing that they do, and setting both graylog and console logging in a single, low-level function is not one atomic operation.

Reply:
It may just be an issue with the name, but I see no issue in creating a function to instantiate a composed logger, as that is a unit of work in itself, instantiating a certain type of logger, no? 

Having a `SetGELFLogger` and `SetConsoleLogger` to me is more confusing because it gives the impression you would be overwriting the logger? But if that is what you want I can give it a whirl.

Reply:
Would `enableGELFLogger` and `enableConsoleLogger` be better? That way it would look something like this

```go

func initLogger(cfg Config) error {
       var zapcore zap.Core
	if cfg.getIsTestEnv() {
		return setLogger(enableTestLogger())
	}

	// Check if graylog host is defined.
	graylogHost := cfg.getGraylogHost()
	if graylogHost != "" {
		zapcore = enableGELFLogger(zapCore)
	}

        zapcore = enableConsoleLogger(zapCore)

	return setLogger(zapCore)
}
```

Reply:
To me it should be like this,

`initLoggers` -> `setConsoleLogger(); if GRAYLOG_HOST then setGELFLogger` (pseudocode, obv)

Reply:
So I think we're in agreement, save for the naming differences.

Reply:
lgtm

Reply:
@ian-axelrod I had to make a few changes due to some constraints on how we instantiate the GELF logger, but let me know what you think.